### DISCUSSION TITLE
or, __or__, and, __and__and the difference between Int32 and int

### DISCUSSION THREAD

**Discussion by RtyLiu** [modular/modular#1808] [Q&A] [answered]
I do not understand logic hidden here, especially the difference between int and Int 32. Execute this and see:
```mojo
fn main():
    var x1: Int32 = 0
    var y1: Int32 = 5
    var z1: Int32 = 6
    var x2 = 0
    var y2 = 5
    var z2 = 6

    print(x1 or z1, x2 or z2)
    print(x1.__or__(z1), x2.__or__(z2))
    print(x1 and z1, x2 and z2)
    print(x1.__and__(z1), x2.__and__(z2))
    print(y1 or z1, y2 or z2)
    print(y1.__or__(z1), y2.__or__(z2))
    print(y1 and z1, y2 and z2)
    print(y1.__and__(z1), y2.__and__(z2))
```

Result:
```
0 6
6 6
6 0
0 0
6 5
7 7
5 6
4 4
```

P.S. I know why (give this result "alias Int32 = SIMD[si32, 1]", but I do not know why (should be a difference) :)

**Comment by soraros** [comment, marked as answer]
Cause of the problem is [this](https://github.com/modularml/mojo/issues/1623#issuecomment-1892491326). Mojo got `bool(Int32(...))` etc. flipped, it's horrible.

**Reply by RtyLiu** [reply]
Thanks.  :)
